webapp/packages/core-root/src/SessionResource.ts

```diff
@@ -95,6 +92,23 @@ export class SessionResource extends CachedDataResource<SessionState | null> {
     return session;
   }
 
+  touchSession = () => {
+    if (this.touchSessionTimer) {
+      return;
+    }
+
+    if (this.data?.valid) {
+      this.graphQLService.sdk.touchSession();
+    }
+
+    this.touchSessionTimer = setTimeout(() => {
+      if (this.touchSessionTimer) {
+        clearTimeout(this.touchSessionTimer);
+        this.touchSessionTimer = null;
+      }
+    }, SESSION_TOUCH_TIME_PERIOD);
```

Comment:
please move it to `SessionExpireService`
so:
in `SessionExpireService` you will run interval and check if user active or not and if it is active you will fire `touchSession`

Reply:
This leads to cyclic dependency. My solution of this did not bring us any fine result. I also had several tries to reduce it manually and find the place where something went wrong, but it feels like there is no easy peasy way to reduce this problem without causing a possible new ones 

I decided to move logic back to the resource so we have no cyclic dependency. I believe this problem deserves a tech ticket